Updated iServices tutorial for Q1 2021.

sickcodes opened the pull request:
Happy to start again on a clean slate with a more appropriate submission.

Let me know if needs any changes!

khronokernel commented on universal/iservices.md:
```diff
@@ -1,10 +1,119 @@
-## Generate a new Serial
+## Generate a new Serial and Board Serial (MLB) for your model.
+
+Compile [macserial](https://github.com/acidanthera/OpenCorePkg/tree/master/Utilities/macserial) from source.
```
Instead of saying compile for all users, mention Linux users must compile. Otherwise, simply use release to avoid issues

sickcodes replied:
I only now noticed the release download, was it always there?

sickcodes replied:
Ah the problem with latest is I don't know how to refer to latest.tar.gz I think that's why I compiled it

vit9696 replied:
@khronokernel maybe we should make utilities compile on Linux as well? Could you and @dhinakg look at it? Will require a Linux C toolchain running in macOS, but that should be easier than I think. I have some scripts on how to make it if necessary.

khronokernel replied:
@sickcodes I would till like this updated for users to use release by default and mention edge case for Linux users that require compile

khronokernel commented on universal/iservices.md:
```diff
@@ -18,7 +127,7 @@ We’re sorry, but this serial number isn’t valid |  Valid Purchase date | Pur
 ![](../images/post-install/iservices-md/not-valid.png) | ![](../images/post-install/iservices-md/valid.png) |  ![](../images/post-install/iservices-md/no-purchase.png)
 
-This last one is what we're after, as we want something genuine but currently not in use by anyone. Now we can translate the rest of the values into our config.plist -> PlatformInfo -> Generic:
+This first one is what we're after, as we want something not in use by anyone. Now we can translate the rest of the values into our config.plist -> PlatformInfo -> Generic:
```
As mentioned here: [Как завести сервисы Apple - iMessage, FaceTime, iCloud](https://applelife.ru/posts/727913)

Valid serial not in use is ideal

sickcodes replied:
The problem with using the third one is that it's unethical. I've found now they all work actually and this is one of the outdated steps I was referring to earlier. All of my devices are using invalid serials. I don't believe the support tool actually affects being able to log in. Happy to test further tho!

hatf0 replied:
Where did you get "unethical" from? If it's a fully valid (as in, legitimate Mac) serial, then IIRC, you wouldn't be able to activate it with iCloud in the first place. I see no place where "ethics" plays a role here.

sickcodes replied:
Yeah you can activate all of them, same as when I use two VMs with the same serial. Or if someone sells their MacBook. A Catalina serial becomes a Big Sur. And opening Catalina again creates another device.

Or did you mean activating an iCloud locked/stolen device?

Regardless, for iMessage, all of them work.

But to get the write answer, I'll test App Store for each type of serial this week and confirm. I've never had issues logging into the App Store so far (KVM). Dataenters, VPNs, different countries. It all just works.

The Apple support tool honestly seems unrelated anymore to activating serials, I believe it's just for this style warranty: ![image](https://user-images.githubusercontent.com/65906298/112074055-475e7180-8b6d-11eb-9729-d97c3b394352.jpeg)

khronokernel replied:
Point still stands, Vitaly can discuss more from this thread if you want: [Как завести сервисы Apple - iMessage, FaceTime, iCloud](https://applelife.ru/posts/727913)

Google Translate should get the point across in this forum, so don't need to be a native speaker

khronokernel commented on universal/iservices.md:
```diff
@@ -1,10 +1,119 @@
+    <key>MLB</key>
+    <string>C02733401J9JG36A8</string>
+    <key>ROM</key>
+    <data>0016cb001122</data>
```
You're using XML but have `<date>` in hexadecimal form instead of base64, would make more sense to have this as a table to not confuse users

sickcodes replied:
Added table: sickcodes@d61e7ea

mhaeuser commented:
This does not make clear why an approach with several manual steps is preferable over GenSMBIOS. If you provide alternatives to an end-user, he needs to be made aware of the considerations of picking one over another.

I agree the process should be automated as you proposed earlier, just not that the automation should live in a remote repository. If your approach has sufficient benefits, it can be upstreamed to OpenCorePkg/macserial, but it will need some defending. To get accreditation straight, it can carry your copyright, it (just) cannot carry links to your pages.

khronokernel commented on universal/iservices.md:
```diff
@@ -1,10 +1,119 @@
+
+If you see a [support warning, see below](#customer-code-error).
+
+### Alternative Option: https://github.com/corpnewt/GenSMBIOS
```
Why is GenSMBIOS 2nd? Most users troubleshooting iServices will have already installed and others will be on macOS or Windows. The Linux user base is tiny with hackintoshes compared to other platforms

Would be better to have GenSMBIOS as 1st since it has a friendlier TUI interface

sickcodes replied:
Added table: https://github.com/sickcodes/OpenCore-Post-Install/commit/d61e7ea45af2661913cec449e6b05ab7d5553fe1

sickcodes replied:
Rearranged: https://github.com/sickcodes/OpenCore-Post-Install/commit/4ad07dc2b7a4b55646aa6c1184663e7dc72eb8ac

khronokernel commented on universal/iservices.md:
````diff
@@ -1,10 +1,119 @@
+    <string>976AA603-75FC-456B-BC6D-9011BFB4968E</string>
+```
+
+NOTE: If you have trouble using the App Store, you [may need to fix En0](#fixing-en0), depending on your hardware setup.
````
You forgot to mention what values from Apple's site the user wants, ie. Invalid, Valid, in use

sickcodes replied:
I honestly don't check them anymore because I believe it doesn't matter: all 3 types of support validity work.

sickcodes replied:
In my account right now, I have two invalid serials, and a third one which has a valid purchase date.

Both of the invalid serial numbers are running Big Sur, and using iServices with no issues.

khronokernel replied:
How the page is formatted now, doesn't require you to add an explicit serial check mention. So resolved

khronokernel reviewed (changes requested):
Overall PR's fine, seems a bit much enforcing the longer path at first and only using GenSMBIOS as an alternative. Besides that, I've made the comment throughout. Appreciate the changes

sickcodes commented:
> This does not make clear why an approach with several manual steps is preferable over GenSMBIOS. If you provide alternatives to an end-user, he needs to be made aware of the considerations of picking one over another.
> 
> I agree the process should be automated as you proposed earlier, just not that the automation should live in a remote repository. If your approach has sufficient benefits, it can be upstreamed to OpenCorePkg/macserial, but it will need some defending. To get accreditation straight, it can carry your copyright, it (just) cannot carry links to your pages.

@mhaeuser sure, I can easily write a de-branded one. There's more than just the script in the repo, its mainly for VM's but one of the scripts does wrap macserial. I'll submit a PR or two to macserial! 

I also added some acknowledgements to the main repo: https://github.com/sickcodes/Docker-OSX#author. I do see that the first commit WAS using OpenCore so I've added that, I must've only been locally using Clover prior :)

khronokernel commented on universal/iservices.md:
```diff
@@ -1,24 +1,148 @@
 **Note**: You and you alone are responsible for your AppleID, read the guide carefully and take full responsibility if you screw up. Dortania and other guides are not held accountable for what **you** do.
 
-## Generate a new Serial
+## Using GenSMBIOS: https://github.com/corpnewt/GenSMBIOS
```
Adding a full link in the subheaders looks funky on VuePress and has line wrap issues on mobile, either make it a hyperlink or just remove it (as there's already a link right below to download)

khronokernel commented on universal/iservices.md:
```diff
@@ -1,24 +1,148 @@
 
 ![](../images/post-install/iservices-md/serial-list.png)
 
+## Using macserial: https://github.com/acidanthera/OpenCorePkg/tree/master/Utilities/macserial
```
Same as the the subheader link

khronokernel commented:
As mentioned, just the 3 main comments I still have:

* ~~Full links in subheaders don't play nicely with the VuePress backend~~
  * Just saw you updated that
* Undo change regarding what serial's ideal
* Mention users to use release variant of macserial and only compile for Linux

With those ~~3~~ 2 minor complaints, should be good to merge. Thanks for spending the time on this PR

sickcodes commented:
Should be all resolved now :)

sickcodes commented:
If any information changes in the future RE: iServices, will update as appropriate

khronokernel commented:
Looks good to me, thanks for the PR
